compiler/rustc_target/src/abi/call/sparc64.rs

```diff
@@ -79,17 +79,15 @@ where
     C: HasDataLayout,
 {
     data = arg_scalar(cx, &scalar1, offset, data);
-    if scalar1.value == abi::F32 {
-        offset += Reg::f32().size;
-    } else if scalar2.value == abi::F64 {
```

Comment:
?

Reply:
That was a whole if/else collection that could just as well have been a match

Reply:
Ah no

Reply:
Should've paid more attention, thanks! That's the bug I introduced